pkg/client/config_source.go

```diff
@@ -0,0 +1,171 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	gen "github.com/hashicorp-dev-advocates/waypoint-client/pkg/waypoint"
+)
+
+// ConfigSourceConfig is the configuration for a ConfigSourceConfig. It does stutter
+// but it confines to the naming we've been using for these configs.
+type ConfigSourceConfig struct {
+	// SourceType controls the type, ex: vault.
+	// This field is case sensitive
+	SourceType string
+
+	// Workspace that this config source belongs to. The default is 'default'
+	Workspace string
+
+	// Project to apply the config source to. This field must be set if the
+	// scope is 'project' or 'app'
+	Project string
+
+	// Application to specify the config source to. Required if scope is 'app'
+	Application string
+
+	// Scope of the config source. Valid values are 'global', 'project', and
+	// 'app'. The default is 'global'
+	Scope string
+
+	// Config is a key/value map of configuation to apply to the scoped config
+	// source
+	Config map[string]string
+
+	// hash is an internal value. TODO: Maybe not have this at all
+	hash uint64
+}
+
+func DefaultConfigSourceConfig() ConfigSourceConfig {
+	return ConfigSourceConfig{}
+}
+
+// SetConfigSource sets the config source information for a given scope and
+// workspace
+func (c *waypointImpl) SetConfigSource(
+	ctx context.Context,
+	cfg ConfigSourceConfig,
+) error {
+	return c.setConfigSource(ctx, cfg, false)
```

Comment:
Would it make sense to just pull in the `setConfigSource` logic into this exported func to avoid and extra func call?

Reply:
ohh, I see the false now... as to why you added an extra func call 👍🏻